### PR TITLE
[#1420] fix(client): reportShuffleWriteFailure failed because of IndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by rickyma**
### What changes were proposed in this pull request?

To fix IndexOutOfBoundsException.

### Why are the changes needed?

For [#1420](https://github.com/apache/incubator-uniffle/issues/1420)

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Rerun successfully.

### CODE REVIEW

**Comment by rickyma** [comment]
PTAL @jerqi @yl09099

File: client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java [hunk 1]
```diff
@@ -330,12 +330,15 @@ public boolean incPartitionWriteFailure(
                 });
             List<Map.Entry<String, AtomicInteger>> list =
                 new ArrayList(shuffleServerFailureRecordCount.entrySet());
-            Collections.sort(list, (o1, o2) -> (o1.getValue().get() - o2.getValue().get()));
-            Map.Entry<String, AtomicInteger> shuffleServerInfoIntegerEntry = list.get(0);
-            if (shuffleServerInfoIntegerEntry.getValue().get()
-                > shuffleManager.getMaxFetchFailures()) {
-              shuffleManager.addFailuresShuffleServerInfos(shuffleServerInfoIntegerEntry.getKey());
-              return true;
+            if (list.size() > 0) {
```
Comment (jerqi) [inline client-spark/common/src/main/java/org/apache/uniffle/shuffle/manager/ShuffleManagerGrpcService.java:333]:
```
!list.isEmpty()
```

**Comment by jerqi** [comment]
Could you add a ut for this pull request?

**Comment by rickyma** [comment]
> Could you add a ut for this pull request?

Done. @jerqi

**Review by jerqi — APPROVED** [review]
LGTM, thanks @rickyma , merged to master. cc @yl09099

**Comment by yl09099** [comment]
> LGTM, thanks @rickyma , merged to master. cc @yl09099

LGTM.

**Pull request merged**